client-ts/Microsoft.AspNetCore.SignalR.Client.TS/IHubProtocol.ts

```diff
@@ -28,8 +28,14 @@ export interface NegotiationMessage {
     readonly protocol: string;
 }
 
+export const enum ProtocolType {
+    Text = 1,
+    Binary
+}
+
 export interface IHubProtocol {
     name(): string;
+    type(): ProtocolType;
```

Comment:
You can do `readonly` properties but they are only `readonly` in TypeScript but not in Javascript.